Mathlib/AlgebraicGeometry/Morphisms/FiniteType.lean

```diff
@@ -67,4 +68,40 @@ instance locallyOfFiniteType_isStableUnderBaseChange :
     MorphismProperty.IsStableUnderBaseChange @LocallyOfFiniteType :=
   HasRingHomProperty.isStableUnderBaseChange RingHom.finiteType_isStableUnderBaseChange
 
+instance {R} [CommRing R] [IsJacobsonRing R] : JacobsonSpace (Spec (.of R)) :=
+  inferInstanceAs (JacobsonSpace (PrimeSpectrum R))
+
+instance {R : CommRingCat} [IsJacobsonRing R] : JacobsonSpace (Spec R) :=
+  inferInstanceAs (JacobsonSpace (PrimeSpectrum R))
+
+nonrec lemma LocallyOfFiniteType.jacobsonSpace
+  (f : X ⟶ Y) [LocallyOfFiniteType f] [JacobsonSpace Y] : JacobsonSpace X := by
+  wlog hY : ∃ S, Y = Spec S
+  · rw [jacobsonSpace_iff_of_iSup_eq_top
+      (Scheme.OpenCover.iSup_opensRange (Y.affineCover.pullbackCover f))]
+    intro i
+    have inst : LocallyOfFiniteType (Y.affineCover.pullbackHom f i) :=
+      MorphismProperty.pullback_snd _ _ inferInstance
+    have inst : JacobsonSpace Y := ‹_› -- why is this needed?
```

Comment:
This seems to be a bug. I reduced it to this for now, but will investigate further:
```
import Mathlib.Tactic.WLOG

class Foo (X : Type) : Prop where

set_option trace.Meta.synthInstance true

example {X Y : Type} [Foo Y] : Foo X := by
  wlog hY : 3 = 3
  · have : Foo Y := inferInstance
    sorry
  sorry
```